Mathlib/SetTheory/Cardinal/Basic.lean

```diff
@@ -924,6 +942,24 @@ def prod {ι : Type u} (f : ι → Cardinal) : Cardinal :=
 theorem mk_pi {ι : Type u} (α : ι → Type v) : #(∀ i, α i) = prod fun i => #(α i) :=
   mk_congr <| Equiv.piCongrRight fun _ => outMkEquiv.symm
 
+theorem mk_pi_congr {ι ι' : Type u} {f : ι → Type v} {g : ι' → Type v} (e : ι ≃ ι')
+    (h : ∀ i, #(f i) = #(g (e i))) : #(Π i, f i) = #(Π i, g i) :=
+  mk_congr <| Equiv.piCongr e fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)
+
+theorem mk_pi_congr' {ι : Type u} {ι' : Type v} {f : ι → Type max w (max u v)}
+    {g : ι' → Type max w (max u v)} (e : ι ≃ ι')
+    (h : ∀ i, #(f i) = #(g (e i))) : #(Π i, f i) = #(Π i, g i) :=
+  mk_congr <| Equiv.piCongr e fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)
+
+theorem mk_pi_congrRight {ι : Type u} {f g : ι → Type v} (h : ∀ i, #(f i) = #(g i)) :
+    #(Π i, f i) = #(Π i, g i) :=
+  mk_pi_congr (Equiv.refl ι) h
+
+theorem mk_pi_congr_subtype {ι : Type u} {f g : ι → Type v} {p : ι → Prop}
+    (h : ∀ i, p i → #(f i) = #(g i)) : #(Π i, p i → f i) = #(Π i, p i → g i) := mk_congr <|
+  Equiv.piCongr (Equiv.refl ι)
+    fun i ↦ Equiv.piCongr (.refl _) (fun hi ↦ Classical.choice <| Cardinal.eq.mp (h i hi))
```

Comment:
I don't think this lemma is a good one, it can be proved in terms of the one above and something that you're missing:
```lean
theorem mk_pi_congrRight_prop {ι : Prop} {f g : ι → Type v} (h : ∀ i, #(f i) = #(g i)) :
    #(Π i, f i) = #(Π i, g i) :=
  mk_congr <| Equiv.piCongr (.refl _) fun i ↦ Classical.choice <| Cardinal.eq.mp (h i)

theorem mk_pi_congrRight_subtype {ι : Type u} {f g : ι → Type v} {p : ι → Prop}
    (h : ∀ i, p i → #(f i) = #(g i)) : #(Π i, p i → f i) = #(Π i, p i → g i) := by
  refine mk_pi_congrRight fun i => ?_
  refine mk_pi_congrRight_prop fun hi => ?_
  apply h _ hi
```

Reply:
I'm not sure I understand. I renamed things, now there are `mk_pi_congr`, `mk_pi_congr'`, `mk_pi_congrRight` and `mk_pi_congrRight_prop` (renamed from `_subtype`). What you wrote as `mk_pi_congrRight_prop` is just `mk_pi_congrRight`, and your `mk_pi_congrRight_subtype` is `mk_pi_congrRight_prop`.

Reply:
I also don't follow your comment @eric-wieser, could you elaborate?

Reply:
I made a typo in the above, the difference is `f g : ι → Prop`